opaque_keys/edx/tests/test_asset_locators.py

```diff
@@ -93,3 +93,13 @@ def test_empty_path(self):
             '/c4x/org/course/asset/',
             unicode(CourseKey.from_string('org/course/run').make_asset_key('asset', ''))
         )
+
+    @ddt.data(
+        "asset-v1:UOG+cs_34+Cs128+type@asset+block@subs_Introduction%20To%20New.srt.sjson",
+    )
```

Comment:
You should use this ddt test to test *all* special characters, like ~, :, -, etc.